### PR TITLE
Ping-pong services to help tracing tests

### PR DESCRIPTION
**Pull request by crufter**
Output:

```
crufter@cruftop:~/code/src/github.com/micro/platform(traces)$ micro call go.micro.ping Ping.Ping '{}'{
	"ping": "Ping service called Pong and that responded: Pong"
}
```

Not a tipical ping pong example so ping is doing Ping Ping and pong is doing Pong Pong... haha

### CODE REVIEW

File: ping/go.mod [hunk 1]
```diff
@@ -0,0 +1,10 @@
+module github.com/micro/services/ping
+
+go 1.13
+
+require (
+	github.com/golang/protobuf v1.3.3
+	github.com/micro/cli/v2 v2.1.1
+	github.com/micro/go-micro/v2 v2.0.0
+	github.com/micro/services/pong v0.0.0-00010101000000-000000000000
```
Comment (jakexks) [inline ping/go.mod]: I think you need to change this after merging

File: pong/main.go [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+	fmt.Println(rsp.Pong)
+}
+
+func main() {
+	// Create a new service. Optionally include some options here.
+	service := micro.NewService(
+		micro.Name("go.micro.pong"),
+		micro.Version("latest"),
+		micro.Metadata(map[string]string{
+			"type": "helloworld",
```
Comment (asim) [inline pong/main.go]: Can probably remove Verison and Metadata

File: pong/main.go [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+
+type Pong struct{}
+
+func (g *Pong) Pong(ctx context.Context, req *proto.Request, rsp *proto.Response) error {
+	rsp.Pong = "Pong"
+	return nil
+}
+
+// Setup and the client
+func runClient(service micro.Service) {
```
Comment (asim) [inline pong/main.go]: Can remove this entire method

File: pong/main.go [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+		micro.Version("latest"),
+		micro.Metadata(map[string]string{
+			"type": "helloworld",
+		}),
+
+		// Setup some flags. Specify --run_client to run the client
+
+		// Add runtime flags
+		// We could do this below too
+		micro.Flags(&cli.BoolFlag{
```
Comment (asim) [inline pong/main.go]: Can remove this flag

File: ping/main.go [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+func main() {
+	// Create a new service. Optionally include some options here.
+	service := micro.NewService(
+		micro.Name("go.micro.ping"),
+		micro.Version("latest"),
+		micro.Metadata(map[string]string{
+			"type": "helloworld",
+		}),
+
+		micro.Flags(&cli.BoolFlag{
```
Comment (asim) [inline ping/main.go]: Can remove Flags, Version and Metadata

File: ping/main.go [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+	response := &pongproto.Response{}
+	if err := g.service.Client().Call(ctx, request, response); err != nil {
+		return err
+	}
+	rsp.Ping = "Ping service called Pong and that responded: " + response.GetPong()
+	return nil
+}
+
+// Setup and the client
+func runClient(service micro.Service) {
```
Comment (asim) [inline ping/main.go]: Can remove this function

File: ping/main.go [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+/*
+Example usage of top level service initialisation
+*/
+
+type Ping struct {
+	service micro.Service
+}
+
+func (g *Ping) Ping(ctx context.Context, req *proto.Request, rsp *proto.Response) error {
+	request := client.NewRequest("go.micro.pong", "Pong.Pong", &pongproto.Request{})
```
Comment (asim) [inline ping/main.go]: This should be g.service.Client().NewRequest

File: ping/go.mod [hunk 8]
```diff
@@ -4,7 +4,9 @@ go 1.13
...
 require (
 	github.com/golang/protobuf v1.3.3
-	github.com/micro/cli/v2 v2.1.1
-	github.com/micro/go-micro/v2 v2.0.0
+	github.com/micro/cli/v2 v2.1.2-0.20200203150404-894195727d9c
+	github.com/micro/go-micro/v2 v2.0.1-0.20200211112716-d1d6eada9810
 	github.com/micro/services/pong v0.0.0-00010101000000-000000000000
 )
+
+// replace github.com/micro/services/pong => /home/crufter/code/src/github.com/micro/services/pong
```
Comment (asim) [inline ping/go.mod]: Probably want to remove this from PR

**Pull request merged**